Library/Homebrew/extend/pathname.rb

```diff
@@ -165,13 +165,27 @@ def atomic_write(content)
     # The enclosing `mktmpdir` and the `chmod` are a workaround
     # for https://github.com/rails/rails/pull/34037.
     Dir.mktmpdir(".d", dirname) do |tmpdir|
-      begin
-        FileUtils.chmod "+t", dirname
-      rescue Errno::EPERM
-        :ignore
+      stat = File.stat(dirname)
+      should_fix_sticky_bit = stat.world_writable? && !stat.sticky?
+      if should_fix_sticky_bit
+        begin
+          FileUtils.chmod "+t", dirname
+        rescue Errno::EPERM
```

Comment:
The `#world_writable?` check makes me think there shouldn't be a permission error to `rescue` now, am I missing something?